pkg/utils/validator/clusterresourceplacement_test.go

```diff
@@ -1036,3 +1040,78 @@ func TestIsPlacementPolicyUpdateValid(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateToleration(t *testing.T) {
+	tests := map[string]struct {
+		tolerations []placementv1beta1.Toleration
+		wantErr     error
+	}{
+		"valid tolerations": {
+			tolerations: []placementv1beta1.Toleration{
+				{
+					Key:      "key1",
+					Operator: corev1.TolerationOpExists,
+					Value:    "value1",
+					Effect:   corev1.TaintEffectNoSchedule,
+				},
+				{
+					Key:      "key2",
+					Operator: corev1.TolerationOpExists,
+					Effect:   corev1.TaintEffectNoSchedule,
+				},
+				{
+					Operator: corev1.TolerationOpExists,
+				},
+			},
+			wantErr: nil,
+		},
+		"invalid toleration, key is empty, operator is Equal": {
```

Comment:
add a test case, operator is omitted.

Reply:
add a test case, value contains special characters

Reply:
If toleration operator is "" or TaintEffect is "" kubebuilder will complain
![image](https://github.com/Azure/fleet/assets/17449572/a2a2c2c5-96f9-4d1c-9bc8-e7e93fedc802)